tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -671,6 +654,7 @@ DECLARE_CONNECTION_REDIRECTION_V6_TEST_GROUP("dual_ipv6", socket_family_t::IPv6,
 int
 main(int argc, char* argv[])
 {
+    test_globals_t _globals;
```

Comment:
I think this is incorrect. Instead of using a global variable, this PR is creating a stack variable and initializing it (in line 662). The actual tests will never be able to access this stack variable. All the tests should ideally fail in that case, and it looks like that is happening right now. 

Example failure:
https://github.com/microsoft/ebpf-for-windows/actions/runs/5756053268/job/15605278098

```
2023-08-03T22:39:38.2188711Z ===============================================================================
2023-08-03T22:39:38.2189145Z test cases:  80 |  0 passed | 80 failed
2023-08-03T22:39:38.2189817Z assertions: 101 | 21 passed | 80 failed
```



It looks like the tests are being marked as PASSED even if every test failed due to a different bug (Issue #2707)